x-pack/plugins/data_visualizer/public/application/common/components/top_values/top_values.tsx

```diff
@@ -47,76 +46,54 @@ function getPercentLabel(docCount: number, topValuesSampleSize: number): string
 
-  const { fieldFormats } = data;
-
   if (stats === undefined || !stats.topValues) return null;
-  const {
-    topValues,
-    topValuesSampleSize,
-    count,
-    isTopValuesSampled,
-    fieldName,
-    sampleCount,
-    topValuesSamplerShardSize,
-  } = stats;
-
-  const totalDocuments = stats.totalDocuments;
+  const { topValues, fieldName, sampleCount } = stats;
 
-  const progressBarMax = isTopValuesSampled === true ? topValuesSampleSize : count;
+  const totalDocuments = stats.totalDocuments ?? 0;
+  const topValuesOtherCountPercent =
+    1 - (topValues ? topValues.reduce((acc, bucket) => acc + bucket.percent, 0) : 0);
+  const topValuesOtherCount = Math.floor(topValuesOtherCountPercent * (sampleCount ?? 0));
 
-  const topValuesOtherCount =
-    (progressBarMax ?? 0) -
-    (topValues ? topValues.map((value) => value.doc_count).reduce((v, acc) => acc + v, 0) : 0);
-
-  const countsElement =
-    totalDocuments !== undefined ? (
-      <EuiText color="subdued" size="xs">
-        {isTopValuesSampled ? (
-          <FormattedMessage
-            id="xpack.dataVisualizer.dataGrid.field.topValues.calculatedFromSampleRecordsLabel"
-            defaultMessage="Calculated from {sampledDocumentsFormatted} sample {sampledDocuments, plural, one {record} other {records}}."
-            values={{
-              sampledDocuments: sampleCount,
-              sampledDocumentsFormatted: (
-                <strong>
-                  {fieldFormats
-                    .getDefaultInstance(KBN_FIELD_TYPES.NUMBER, [ES_FIELD_TYPES.INTEGER])
-                    .convert(sampleCount)}
-                </strong>
-              ),
-            }}
-          />
-        ) : (
-          <FormattedMessage
-            id="xpack.dataVisualizer.dataGrid.field.topValues.calculatedFromTotalRecordsLabel"
-            defaultMessage="Calculated from {totalDocumentsFormatted} {totalDocuments, plural, one {record} other {records}}."
-            values={{
-              totalDocuments,
-              totalDocumentsFormatted: (
-                <strong>
-                  {fieldFormats
-                    .getDefaultInstance(KBN_FIELD_TYPES.NUMBER, [ES_FIELD_TYPES.INTEGER])
-                    .convert(totalDocuments ?? 0)}
-                </strong>
-              ),
-            }}
-          />
-        )}
-      </EuiText>
-    ) : (
-      <EuiText size="xs" textAlign={'center'}>
+  const countsElement = (
+    <EuiText color="subdued" size="xs">
+      {totalDocuments > (sampleCount ?? totalDocuments) ? (
         <FormattedMessage
-          id="xpack.dataVisualizer.dataGrid.field.topValues.calculatedFromSampleDescription"
-          defaultMessage="Calculated from sample of {topValuesSamplerShardSize} documents per shard"
+          id="xpack.dataVisualizer.dataGrid.field.topValues.calculatedFromSampleRecordsLabel"
+          defaultMessage="Calculated from {sampledDocumentsFormatted} sample {sampledDocuments, plural, one {record} other {records}}."
```

Comment:
In the file data viz expanded rows, the doc count here always seems to be `0`:

<img width="789" alt="image" src="https://user-images.githubusercontent.com/7405507/201964267-09458f2c-243c-4a24-961a-8a6e15a78f77.png">

Reply:
Fixed here [`7d7ba42` (#144646)](https://github.com/elastic/kibana/pull/144646/commits/7d7ba42c7872758583de72b127d08a9ea731696d)